src/tools.ts

```diff
@@ -41,8 +42,26 @@ function detectInfectionCommand(): string {
     return 'phpdbg -qrr ./vendor/bin/infection';
 }
 
-export default function createTools(config: Config): Array<Tool> {
-    return [
+function backwardCompatibilityCheckTool(action: Action): Tool | null {
+    const baseSha1 = action.getBaseBranchSha1();
+
+    if (baseSha1 === null) {
+        return null;
+    }
+
+    return {
+        // @TODO need to `git fetch baseSha1` from source repo!
```

Comment:
Yep, we need a `git fetch` here! 

Unsure how to handle it best, but @basz ran into some `git clone` problems with private repos, so perhaps we can fix that together with it :)

Reply:
maybe having multiple before scripts might help?

1. `git remote add upstream <upstreamUri>`
2. `git fetch upstream <baseRef>`

Reply:
Might be helpful once this PR is being continued:

https://github.com/symfony/symfony/blob/3dfec48b454e24ad2ee3aace4fb2f05c2109d3c5/.github/workflows/psalm.yml#L32-L35

Reply:
I'll leave this here: 
https://github.com/laminas/laminas-continuous-integration-action/blob/fb9164e3cf7fbbd9abacd4d7f7a4a30f165bb7dd/entrypoint.sh#L78-L79

Seems that the CI Container is already checking out the target branch and thus, should be available